src/lib/common/sol-platform-linux-common.c

```diff
@@ -263,3 +307,581 @@ sol_platform_impl_get_os_version(char **version)
+    SOL_INT_CHECK(err, < 0, err);
+
+    err = mount(dev, mpoint, fstype, flags, opts);
+    if (err != 0) {
+        SOL_ERR("Couldn't mount %s to %s - %s", dev, mpoint,
+            sol_util_strerrora(errno));
+        return -errno;
+    }
+
+    return err;
+}
+
+static void
+on_mount_fork(void *data)
+{
+    int err;
+    struct mount_data *mdata = data;
+
+    err = run_mount(mdata->dev, mdata->mpoint, mdata->fstype, mdata->flags, mdata->opts);
+    sol_platform_linux_fork_run_exit(err);
+}
+
+static void
+on_mount_fork_exit(void *data, uint64_t pid, int status)
+{
+    struct mount_data *mdata = data;
+
+    mdata->async_cb(mdata->dev, mdata->mpoint, mdata->data, pid, status);
+    free(mdata->opts);
+    free(mdata->fstype);
+    free(mdata->mpoint);
+    free(mdata->dev);
+    free(mdata);
+}
+
+int
+sol_platform_impl_mount(const char *dev, const char *mpoint, const char *fstype, unsigned long flags, const char *opts, void (*async_cb)(const char *dev, const char *mpoint, void *data, uint64_t pid, int status), void *data)
+{
+    struct mount_data *mdata;
+
+    SOL_NULL_CHECK(dev, -EINVAL);
+    SOL_NULL_CHECK(mpoint, -EINVAL);
+    SOL_NULL_CHECK(fstype, -EINVAL);
+
+    if (!async_cb)
+        return run_mount(dev, mpoint, fstype, flags, opts);
+
+    mdata = calloc(1, sizeof(struct mount_data));
+    SOL_NULL_CHECK(mdata, -ENOMEM);
+
+    mdata->data = data;
+    mdata->dev = strdup(dev);
+    mdata->mpoint = strdup(mpoint);
+    mdata->fstype = strdup(fstype);
+
+    if (opts)
+        mdata->opts = strdup(opts);
+    
+    mdata->flags = flags;
+    mdata->async_cb = async_cb;
+
+    sol_platform_linux_fork_run(on_mount_fork, on_mount_fork_exit, mdata);
+    return 0;
+}
+
+static void
+sol_uevent_event_dispatch(struct uevent_context *ctx, struct sol_uevent *uevent)
+{
+    uint16_t idx;
+    struct uevent_callback *cb;
+
+    SOL_VECTOR_FOREACH_IDX (&ctx->callbacks, cb, idx) {
+        if ((!cb->action || (cb->action && sol_str_slice_str_eq(uevent->action, cb->action))) &&
+            (!cb->subsystem || (cb->subsystem && sol_str_slice_str_eq(uevent->subsystem, cb->subsystem)))) {
+            cb->uevent_cb((void *)cb->cb_data, uevent);
+        }
+    }
+}
+
+static void
+sol_uevent_read_msg(struct uevent_context *ctx, char *msg, int len)
+{
+    struct sol_uevent uevent = {
+        .modalias = SOL_STR_SLICE_EMPTY,
+        .action = SOL_STR_SLICE_EMPTY,
+        .subsystem = SOL_STR_SLICE_EMPTY,
+        .devtype = SOL_STR_SLICE_EMPTY,
+        .devname = SOL_STR_SLICE_EMPTY,
+    };
+    int i = 0;
+
+    /** lets avoid it to misbehave when people trying to test it with systemd running */
+    if (!strncmp(msg, LIBUDEV_ID, sizeof(LIBUDEV_ID))) {
+        SOL_INF("We're running side-by-side with udevd, skipping udevd generated event");
+        return;
+    }
+
+    while (i < len) {
+        char *curr, *del;
+        size_t value_len;
+
+        curr = msg + i;
+        del = strrchr(curr, '=');
```

Comment:
We are skipping all udev events which by the way is valid only while testing cause it's meant to run without any udev running.